source/Platforms.hx

```diff
@@ -0,0 +1,178 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxObject;
+import flixel.FlxSprite;
+import flixel.system.FlxAssets.FlxGraphicAsset;
+import flixel.util.FlxColor;
+import flixel.input.keyboard.FlxKey;
+import flixel.group.FlxGroup;
+
+/**
+ * ...
+ * @author Jess Geiger
+ */
+class Platforms extends FlxSprite 
+{
+    //set initial movement
+    private var UP:Bool = false;
+    private var DOWN:Bool = true;
+    private var LEFT:Bool = false;
+    private var RIGHT: Bool = true;
+    private var heavy:Float = 8000; // Declare higher gravity while on platform
+    private var blockSize:Int = 32; //Declare block size to be used in width and height
+    //Initialize min and max positioning
+    private var minX:Float;
+    private var maxX:Float;
+    private var minY:Float;
+    private var maxY:Float;
+    //Set starting position and width of platform
+    private var startX:Float;
+    private var startY:Float;
+    private var platformwidth:Int;
+    public var offsetX:Float = 0;
+    public var inContact:Bool = false;
+    private var platformVelocity:Int = 40;
+    public var touchingSprites:FlxTypedGroup<PlatformTracker> = new FlxTypedGroup<PlatformTracker>();
+    private var countTouching:Int = 0;
+
+    public function new(?X:Float=0, ?Y:Float=0, ?W:Int=0, ?L:Float=0, ?R:Float=0, ?U:Float=0, ?D:Float=0, ?trackPlayer:Player, ?SimpleGraphic:FlxGraphicAsset) 
+	{
+		super(X, Y, SimpleGraphic);
+		//set starting positions based on requested dimensions/positions
+        startX = X;
+        startY = Y;
+        minX = L;
+        maxX = R;
+        minY = U;
+        maxY = D;
+        platformwidth = blockSize * W;
+
+		// Initializes a basic graphic for the player
+		makeGraphic(platformwidth, blockSize, FlxColor.YELLOW);
+
+	}
+
+    //Movement as requested by user parameters
+    public function movement():Void
+    {
+		if (DOWN) {
+            if (y >= (startY + maxY)) {
+                UP = true;
+                DOWN = false;
+            }
+            else {
+                velocity.y = platformVelocity;
+            }
+        }
+        if (UP) {
+            if (y <= (startY - minY)) {
+                UP = false;
+                DOWN = true;
+            }
+            else {
+                velocity.y = -platformVelocity;
+            }
+        }
+        if (LEFT) {
+            if (x <= (startX - minX)) {
+                RIGHT = true;
+                LEFT = false;
+            }
+            else {
+                velocity.x = -platformVelocity;
+            }
+        }
+        if (RIGHT) {
+            if (x >= startX + maxX) {
+               RIGHT = false;
+               LEFT = true;
+            }
+            else {
+                velocity.x = platformVelocity;
+            }
+        }
+    }
+
+    // Add or update sprites colliding with platform
+    public function platformObjects(elapsed:Float, player:FlxObject, platform:Platforms):Void
```

Comment:
/**
     *  Add or update sprites colliding with the platform. Add the sprite to the touching platform group if
     *  the group is empty. If it is not empty, iterate through until the sprite is either found (in which
     *  case update its positioning) or add it to the list if it is not already present.
     *  
     *  @param   elapsed is the time passed since last update
     *  @param   player is the sprite object
     *  @param   platform is the platform which the sprite object is touching
     */